src/libraries/System.Net.Http/src/System/Net/Http/DiagnosticsHandler.cs

```diff
@@ -98,12 +114,22 @@ internal static bool IsEnabled()
                 InjectHeaders(currentActivity, request);
             }
 
-            Task<HttpResponseMessage>? responseTask = null;
+            HttpResponseMessage? response = null;
+            TaskStatus taskStatus = TaskStatus.Faulted;
             try
             {
-                responseTask = base.SendAsync(request, cancellationToken);
-
-                return await responseTask.ConfigureAwait(false);
+                if (async)
+                {
+                    Task<HttpResponseMessage>? responseTask = base.SendAsync(request, cancellationToken);
+                    response = await responseTask.ConfigureAwait(false);
+                    taskStatus = responseTask.Status;
```

Comment:
This construction is strange.  If responseTask wasn't yet completed, we'd never get here, and if it completed as anything other than successfully, we'd never get here, so TaskStatus here will always be RanToCompletion.  Which means this whole if/else can be restructued as:
```C#
response = async ?
    await base.SendAsync(request, cancellationToken) :
    base.Send(request, cancellationToken);
taskStatus = TaskStatus.RanToCompletion;
```

Reply:
It doesn't make sense, it's a mistake. I slightly rewritten this part to get the status directly from `responseTask` in case of async (as it was before).